pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper.go

```diff
@@ -51,7 +57,33 @@ func newGitLabScraper(
 	}
 }
 
-// scrape and return metrics
+type branchData struct {
+	ProjectPath string
+	BranchNames []string
+}
+
+// Returns a struct with the project path and an array of branch names via the given channel.
+//
+// The max number of branch names currently returned is 100,000 since the GitLab API will let us get
+// as much as we want but it will take a long time to get all the branches. The biggest repo we
+// found was a GitLab repo with just over 100,000 branches which is an extreme edge case which we
+// believe is not worth supporting.
+func getBranches(
+	ctx context.Context,
+	graphClient graphql.Client,
+	projectPath string,
+	ch chan branchData,
+	waitGroup *sync.WaitGroup,
+) error {
+	defer waitGroup.Done()
+	branches, err := getBranchNames(context.Background(), graphClient, projectPath)
```

Comment:
we set `branches, err` here but call `branches on the next line before returning the `err` up to the calling function. We should check the error here in this call given that if an error is returned, then it's possible that `branches.Project.Repository.BranchNames` won't exist, likely causing a panic.